drv/Map.drv

```diff
@@ -142,7 +210,459 @@ public interface MAP KEY_VALUE_GENERIC extends FUNCTION KEY_VALUE_GENERIC, Map<K
 
 	@Deprecated
 	@Override
-	boolean containsValue(Object value);
+	default boolean containsValue(Object ov) {
+		if (!(ov instanceof VALUE_CLASS)) return false;
+		return containsValue(VALUE_OBJ2TYPE(ov));
+	}
+
+#endif
+
+	// Defaultable methods
+
+#if KEYS_PRIMITIVE || VALUES_PRIMITIVE
+
+	/** Returns the value to which the specified key is mapped, or the {@code defaultValue} if this map contains no mapping for the key.
+	 *
+	 * @param key the key.
+	 * @param defaultValue the default mapping of the key.
+	 *
+	 * @return the value to which the specified key is mapped, or the {@code defaultValue} if this map contains no mapping for the key.
+	 *
+	 * @see java.util.Map#getOrDefault(Object, Object)
+	 * @since 8.0.0
+	 */
+
+	default VALUE_GENERIC_TYPE GET_VALUE_OR_DEFAULT(KEY_TYPE key, VALUE_GENERIC_TYPE defaultValue) {
+#if VALUES_PRIMITIVE
+		return containsKey(key) ? GET_VALUE(key) : defaultValue;
+#else
+		V v;
+		return (v = get(key)) != null || containsKey(key) ? v : defaultValue;
+#endif
+	}
+
```

Comment:
I think that the logic in the following methods does not suit to fastutil. In fastutil, the default return value replaces for all purposes null, even in object-based methods. This has a number of useful consequences: if the user, for instance, ignores it, he/she gets back the JDK behaviour. But I could create a single marker object as default value and then I could use null as a return value without any confusion between existing mappings and mappings to null.

Thus, specification and implementation of the following methods should be uniform: same semantics of the JDK were "null" is replaced by the default value (as it happens currently for primitive types).